domain/src/test/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/CalendarTest.java

```diff
@@ -0,0 +1,664 @@
+
+        verify(mockNoticeCollection, times(1)).clear();
+        verify(mockNoticeCollection, times(1)).add(captor.capture());
+
+        final List<MissingRequiredValueNotice> noticeList = captor.getAllValues();
+
+        assertEquals(FILENAME, noticeList.get(0).getFilename());
+        assertEquals("monday", noticeList.get(0).getFieldName());
+        assertEquals(SERVICE_ID, noticeList.get(0).getEntityId());
+
+        assertTrue(buildResult.getData() instanceof List);
+        verifyNoMoreInteractions(mockNoticeCollection);
+    }
+
+    @Test
+    public void createCalendarWithInvalidTuesdayValueShouldGenerateNotice() {
+        @SuppressWarnings("unchecked") final List<Notice> mockNoticeCollection = mock(ArrayList.class);
+        final Calendar.CalendarBuilder underTest = new Calendar.CalendarBuilder(mockNoticeCollection);
+
+        underTest.serviceId(SERVICE_ID)
+                .monday(0)
+                .tuesday(3)
+                .wednesday(0)
+                .thursday(0)
+                .friday(0)
+                .saturday(0)
+                .sunday(0)
+                .startDate(LocalDateTime.now())
+                .endDate(LocalDateTime.now());
+
+        final EntityBuildResult<?> buildResult = underTest.build();
+
+        final ArgumentCaptor<IntegerFieldValueOutOfRangeNotice> captor =
+                ArgumentCaptor.forClass(IntegerFieldValueOutOfRangeNotice.class);
+
+        verify(mockNoticeCollection, times(1)).clear();
+        verify(mockNoticeCollection, times(1)).add(captor.capture());
+
+        final List<IntegerFieldValueOutOfRangeNotice> noticeList = captor.getAllValues();
+
+        assertEquals(FILENAME, noticeList.get(0).getFilename());
+        assertEquals("tuesday", noticeList.get(0).getFieldName());
+        assertEquals(SERVICE_ID, noticeList.get(0).getEntityId());
+        assertEquals(3, noticeList.get(0).getActualValue());
+        assertEquals(1, noticeList.get(0).getRangeMax());
+        assertEquals(0, noticeList.get(0).getRangeMin());
+
+        assertTrue(buildResult.getData() instanceof List);
+        verifyNoMoreInteractions(mockNoticeCollection);
+    }
+
+    @Test
+    void createCalendarWithNullTuesdayValueShouldGenerateNotice() {
+        @SuppressWarnings("unchecked") final List<Notice> mockNoticeCollection = mock(ArrayList.class);
+        final Calendar.CalendarBuilder underTest = new Calendar.CalendarBuilder(mockNoticeCollection);
+
+        underTest.serviceId(SERVICE_ID)
+                .monday(0)
+                .tuesday(null)
+                .wednesday(0)
+                .thursday(0)
+                .friday(0)
+                .saturday(0)
+                .sunday(0)
+                .startDate(LocalDateTime.now())
+                .endDate(LocalDateTime.now());
+
+        final EntityBuildResult<?> buildResult = underTest.build();
+
+        final ArgumentCaptor<MissingRequiredValueNotice> captor =
```

Comment:
I understand the use of the captor in conjunction with the `CalendarBuilder` parametric constructor taking a list, but reading this test I can't help but wonder if we shouldn't simply unwrap the list from `buildResult` (cast its `data` member to a List). Is there something I missed?

Reply:
As discussed, we will use de default parametric constructor here